test: Edgehub Config Unit test failed

<!-- Please only use this template for submitting reports about failing tests in KubeEdge CI jobs -->

**Which jobs are failing**:
https://travis-ci.org/kubeedge/kubeedge/jobs/508213930

**Which test(s) are failing**:
edgehub config unit test.

**Since when has it been failing**:

**Reason for failure**:
2019/03/19 04:52:10 DEBUG: file event [/tmp/kubeedge/testData REMOVE], operation is %!d(MISSING). reload it.
2019/03/19 04:52:10 WARN: the file change mode: ["/tmp/kubeedge/testData": REMOVE], continue
2019/03/19 04:52:10 DEBUG: watch file error:%!(EXTRA []interface {}=[<nil>])
2019/03/19 04:52:10 DEBUG: file event [/tmp/kubeedge/testData/edge.yaml REMOVE], operation is %!d(MISSING). reload it.
2019/03/19 04:52:10 WARN: the file change mode: ["/tmp/kubeedge/testData/edge.yaml": REMOVE], continue
2019/03/19 04:52:10 DEBUG: file event [/tmp/kubeedge/testData REMOVE], operation is %!d(MISSING). reload it.
2019/03/19 04:52:10 WARN: the file change mode: ["/tmp/kubeedge/testData": REMOVE], continue
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x73f8d3]
goroutine 83 [running]:
github.com/kubeedge/kubeedge/vendor/github.com/ServiceComb/go-archaius/sources/file-source.(*watch).startWatchPool(0xc0003729c0)
	/home/travis/gopath/src/github.com/kubeedge/kubeedge/vendor/github.com/ServiceComb/go-archaius/sources/file-source/yamlconfigurationsource.go:384 +0x73
created by github.com/kubeedge/kubeedge/vendor/github.com/ServiceComb/go-archaius/sources/file-source.(*yamlConfigurationSource).DynamicConfigHandler
	/home/travis/gopath/src/github.com/kubeedge/kubeedge/vendor/github.com/ServiceComb/go-archaius/sources/file-source/yamlconfigurationsource.go:361 +0x91
FAIL	github.com/kubeedge/kubeedge/edge/pkg/edgehub/config	0.019s
ok  	github.com/kubeedge/kubeedge/edge/pkg/edgehub	(cached)
ok  	github.com/kubeedge/kubeedge/edge/pkg/edgehub/clients	0.006s
ok  	github.com/kubeedge/kubeedge/edge/pkg/edgehub/clients/wsclient	300.117s
ok  	github.com/kubeedge/kubeedge/edge/pkg/eventbus/common/util	0.018s
ok  	github.com/kubeedge/kubeedge/edge/pkg/metamanager/dao	0.007s
ok  	github.com/kubeedge/kubeedge/edge/pkg/metamanager	64.101s

**Anything else we need to know**:

--- a/edge/pkg/common/util/config.go
+++ b/edge/pkg/common/util/config.go
@@ -22,7 +22,6 @@ import (
 	"crypto/x509"
 	"crypto/x509/pkix"
 	"encoding/pem"
-	"io/ioutil"
 	"math/big"
 	"os"
 	"path"
@@ -78,8 +77,18 @@ func GenerateTestYaml(test interface{}, path, filename string) error {
 	if err != nil {
 		return err
 	}
-	err = ioutil.WriteFile(path+"/"+filename+".yaml", data, 0777)
+	fp, err := os.Create(path + "/" + filename + ".yaml")
 	if err != nil {
+		return err
+	}
+	_, err = fp.Write(data)
+	if err != nil {
+		return err
+	}
+	if err = fp.Sync(); err != nil {
+		return err
+	}
+	if err = fp.Close(); err != nil {
 		return err
 	}
 	return nil

